### PR TITLE
Error formatting output

### PR DESCRIPTION
**Pull request by liubo0127**
@LinuxGit @superlzs0476 PTAL

### CODE REVIEW

File: callback_plugins/help.py [hunk 1]
```diff
@@ -11,14 +12,19 @@
       - This plugin will print help message when tasks fail.
 '''
 
-from ansible.plugins.callback import CallbackBase
+from ansible.plugins.callback import CallbackBase, strip_internal_keys
+from ansible.module_utils._text import to_bytes, to_text
+from ansible.utils.color import stringc
 from ansible import constants as C
+import locale
```
Comment (zhouqiang-cl) [inline callback_plugins/help.py]: how about move the system import in front of the file

File: callback_plugins/help.py [hunk 2]
```diff
@@ -37,19 +47,98 @@ def __init__(self):
         self.handler = logging.FileHandler(FAIL_LOGFILE)
         self.logger.addHandler(self.handler)
 
+    def format_results(self, result):
+        messages = '[' + str(result._host.name) + ']: Ansible Failed! => changed=' + \
```
Comment (zhouqiang-cl) [inline callback_plugins/help.py]: use format?

File: callback_plugins/yaml.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      - default_callback
+    requirements:
+      - set as stdout in configuration
+'''
+
+import yaml
+import json
+import re
+import string
+import sys
```
Comment (zhouqiang-cl) [inline callback_plugins/yaml.py]: how about add a blank line

File: callback_plugins/help.py [hunk 4]
```diff
@@ -37,19 +47,98 @@ def __init__(self):
...
+
+            try:
+                fileobj.flush()
+            except IOError as e:
+                # Ignore EPIPE in case fileobj has been prematurely closed, eg.
+                # when piping to "head -n1"
+                if e.errno != errno.EPIPE:
+                    raise
+
+        if logger and not screen_only:
```
Comment (qrr1995) [inline callback_plugins/help.py]: When would this  if be true since the `logger`is set to `None` in line 78?

File: callback_plugins/help.py [hunk 5]
```diff
@@ -67,7 +156,8 @@ def v2_playbook_on_stats(self, stats):
                 self._display.banner("ERROR MESSAGE SUMMARY")
                 with io.open(FAIL_LOGFILE, 'r', encoding="utf-8") as f:
                     for _, line in enumerate(f):
```
Comment (qrr1995) [inline callback_plugins/help.py]: Use `for line in f:` is more elegant to read a file line by line(Still enumerated)

File: callback_plugins/yaml.py [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    """Returns true if string should be in block format"""
+    for c in u"\u000a\u000d\u001c\u001d\u001e\u0085\u2028\u2029":
+        if c in value:
+            return True
+    return False
+
+
+def my_represent_scalar(self, tag, value, style=None):
+    """Uses block style for multi-line strings"""
+    if style is None:
```
Comment (qrr1995) [inline callback_plugins/yaml.py]: `if not style`

File: callback_plugins/yaml.py [hunk 7]
```diff
@@ -0,0 +1,126 @@
...
+            # ...tabs prevent blocks from expanding
+            value = value.expandtabs()
+            # ...and odd bits of whitespace
+            value = re.sub(r'[\x0b\x0c\r]', '', value)
+            # ...as does trailing space
+            value = re.sub(r' +\n', '\n', value)
+        else:
+            style = self.default_style
+    node = yaml.representer.ScalarNode(tag, value, style=style)
+    if self.alias_key is not None:
```
Comment (qrr1995) [inline callback_plugins/yaml.py]: Same as above, `if self.alias_key`

**Pull request merged**